### DISCUSSION TITLE
Host an app locally

### DISCUSSION THREAD

**Discussion by letscodetpl** [serverpod/serverpod#2518] [Q&A] [answered]
Hi everyone, I have made a simple app that records visitors check in and check out with serverpod. Next step I am thinking is getting this deploy on a local server, no external access. Just wanted to ask for direction to go or explore, like what tools should be used? what else needs to learn? etc. Thanks a lot.

**Comment by rotoclabs** [comment, marked as answer]
> Just wanted to ask for direction to go or explore, like what tools should be used? what else needs to learn? etc. Thanks a lot.

Have a look at the Dockerfile, it primarily builds the kernel and copies out a few files into a directory structure. It looks to me that you could simply run up that docker container (I hope you would have a running docker engine since the serverpod getting started guides need you to setup up one, unless it it's the mini pod).just replicate those steps for local deployment.

**Reply by letscodetpl** [reply]
@robythomas I appreciate your reply to my question.
What I have tried was using python (python -m http.server {PORT}) to run. Steps I have tried are below
1. run flutter build web
2. run dart bin/main.dart--apply-migrations
3. go to build/web directory run python -m http.server {PORT}

After these steps, I was able to using other devices like other pcs or my phone to access the app, however I was only able to go through UI like clicking on buttons to take to different pages but was not able to read and write to database. Do you have any better suggestion to do the same with the database? Appreciate your help.

**Reply by rotoclabs** [reply]
Your question seems to be asking for two things. 

1. How to setup a PostgreSQL test database for this
2. How to configure the server side app to connect to the test database which you've setup.
 
For (1), it's a topic perhaps beyond this serverpod forum, however https://www.postgresql.org/docs/14/admin.html might be a good place to start at. I earlier assumed you've had a Docker based PostgreSQL running, otherwise you wouldn't have been able to do the development build anyway, and I still believe that's the easiest option. Because it already exposed a port of the container running, on the `localhost` in that configuration.

Now, assuming that you've setup the database instance as described in the doc above, and you are able to connect it using the SQL clients that the PostgreSQL has provided, next is to configure your application to talk to the very same database.

In the `<app>_server/config/development.yaml` (Your run command seems to qualify as development run), specify the

```
database:
  host: localhost
  port: 5432
  name: postgres
  user: postgres
```

and set the appropriate password for this, in the `<app>_server/config/passwords.yaml` which might look like

```
development:
  database: <your db user's password here>
```

With this if you restart the server app, I believe it should connect to this database and proceed. Since this is a development run and you are using the `--apply-migrations` you won't need to do production database preparatory steps like directly running the `definition.sql` or `migration.sql`.
